snap-raster/src/main/java/org/esa/snap/raster/gpf/FilterOperator.java

```diff
@@ -112,87 +104,84 @@ private void populateFilterMap(Filter[] filters) {
     @Override
     public void initialize() throws OperatorException {
         ensureSingleRasterSize(sourceProduct);
+        if (userDefinedKernelFile == null && selectedFilterName == null) {
+            // we cannot throw exception here because, in this case the GraphBuilder is not working anymore
+//            throw new OperatorException("Parameter 'userDefinedKernelFile' or 'selectedFilterName' must be specified");
+            // instead we log an error
+            SystemUtils.LOG.severe("Parameter 'userDefinedKernelFile' or 'selectedFilterName' must be specified");
+        }
+        if (sourceBandNames == null || sourceBandNames.length == 0) {
+            // we cannot throw exception here because, in this case the GraphBuilder is not working anymore
```

Comment:
This should be exceptions. The graphbuilder validation works by initialization of products. 
Logs will not inform the user that the current configuration will not work for that product

Reply:
Yes, unfortunately, the exceptions do not work for the operator because they interrupt the initialisation of the GUI and the operator can't be used anymore in the Graph Builder. As this validation wasn't done before it is at least a minor improvement for the usage with gpt.

I've created a ticket for the exception issue.
https://senbox.atlassian.net/browse/SNAP-1557